ArduPlane/ArduPlane.cpp

```diff
@@ -290,8 +290,8 @@ void Plane::one_second_loop()
     // update home position if NOT armed and gps position has
     // changed. Update every 5s at most
     if (!arming.is_armed() &&
-        gps.last_message_time_ms() - last_home_update_ms > 5000 &&
-        gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
+            gps.status() >= AP_GPS::GPS_OK_FIX_3D &&
```

Comment:
Plane requires GPS for flying at the moment. The net effect of the change is the most common case is to pass the fix type, whereas if you were trying to do a faster processing checking the update time first is likely to reduce the time the most. (But really for what this is this really doesn't help things (or hurt) either way, just noise in the history...

Reply:
It actually doesn't. A lot of these changes are from me doing GPS denied test flights. Even before, during, and after an AUTO landing. As long as you have it once, it can deadreckon pretty damn good

Reply:
If you run all your arming checks we don't let you arm without it. I'd be interested to see logs from dead reckoning if you can share them though, as last time I saw a plane try and do it, it went terribly...

On topic I guess my point is I'm not sure that this is really better in terms of time, or clarity. Just seems like churn, but I'll probably mostly checkout on it here at this point.... (Although I would be interested in seeing the dead reckoning logs)